src/crimson/osd/pg.cc

```diff
@@ -1510,6 +1510,20 @@ void PG::context_registry_on_change() {
   }
 }
 
+bool PG::is_missing_head_and_clones(const hobject_t &hoid) {
```

Comment:
We can do this more efficiently that checking each individually.  pg_missing_set::missing contains all of the missing objects in hobject_t order.  Something like missing.lower_bound(obj.get_object_boundary()) == missing.upper_bound(obj.max_object_boundary()) should evaluate to true iff the missing set contains no head or clones of obj.  I'd suggest adding a helper there (is_missing_any_head_or_clone_of?) and exposing it through PeeringState.

Reply:
Great suggestion! Updated, thanks!